### PR TITLE
[cleanup] pkg/daemon: do not use pointers to strings for errors + unexport consts

### PR DESCRIPTION
**Pull request by runcom**
Pointer to strings in Golang as used rarely and almost only for struct
properties when deserializing json or yaml (or anything) into a structure.
Else, the idiomatic way of informing something isn't right it's through
errors only. Errors in Go do also have an handy `.Error()` which represents
the error as a string for our use cases.

Second commit unexports consts not used outside the daemon package.

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -124,12 +124,12 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 	reconcilableError := dn.reconcilable(oldConfig, newConfig)
 
 	if reconcilableError != nil {
```
Comment (kikisdeliveryservice) [inline pkg/daemon/update.go]: Can we then say `if reconcilableError {` now?

Comment (runcom) [reply pkg/daemon/update.go]: nope, it's still an error (your case it's only valid for `bool` in Go)

Comment (kikisdeliveryservice) [reply pkg/daemon/update.go]: ahh TIL thank you!!

**Comment by cgwalters** [comment]
/lgtm

But...this is all good cleanup but I still think we need to be careful about code churn at this point.

**Comment by runcom** [comment]
> But...this is all good cleanup but I still think we need to be careful about code churn at this point.

I'm trying to be as concise as I can when cleaning up something to avoid regressions and overlooks

**Pull request merged**